caml_z.c

```diff
@@ -934,6 +934,30 @@ CAMLprim value ml_z_extract(value arg, value off, value len)
       if (x >= 0) return Val_long(x);
       /* If x < 0, fall through slow path */
     }
+  } else if (l <= Z_BASE2_LENGTH_OP) {
+    Z_ARG(arg);
+    c1 = o / Z_LIMB_BITS;
```

Comment:
If anyone else reads this and wonders whether it wouldn't be better to convert `o` to an unsigned integer type before computing `/` and `%`, the answer is, “with GCC and Clang at `-O`, yes, although not at `-O2`. See https://gcc.godbolt.org/z/Ya4vxW

I would recommend casting `o` to `uintnat` before these operations.

Reply:
Note that I simply reused what I found below: this part is the same as the slow path. So if needed, it should be changed in both places.
Yet I would have directly used corresponding bitwise operation `>>` and `&` but I though that `/` and `%` were still clearer and that the code would be compiled with optimisation so we would have not to bother with this kind of subtlety.

Reply:
The new version should address this issue.